ethcore/src/client/client.rs

```diff
@@ -2545,16 +2545,6 @@ impl ProvingBlockChainClient for Client {
 
 impl SnapshotClient for Client {}
 
-impl Drop for Client {
-	fn drop(&mut self) {
-		if let Some(c) = Arc::get_mut(&mut self.engine) {
-			c.stop()
```

Comment:
AFAICT `Clique` is the only engine that actually impls `stop()`. And the `Arc::get_mut` here fails every time.

Reply:
Can you change https://github.com/paritytech/parity-ethereum/blob/master/ethcore/src/engines/mod.rs#L429 to `fn stop(&self)` it is broken, because `Engine` is stored as `Arc<Engine>` and doesn't support mutation.

I'm sorry that I didn't catch when reviewing it :(